README.md

```diff
@@ -35,19 +35,20 @@ FLAGS:
     -m, --multipart       Similar to --form, but always sends a multipart/form-data request (i.e., even without files)
     -I, --ignore-stdin    Do not attempt to read stdin
     -F, --follow          Do follow redirects
-    -d, --download
+    -d, --download        Download the body to a file instead of printing it
     -h, --headers         Print only the response headers, shortcut for --print=h
     -b, --body            Print only the response body, Shortcut for --print=b
     -c, --continue        Resume an interrupted download
     -v, --verbose         Print the whole request as well as the response
     -q, --quiet           Do not print to stdout or stderr
     -S, --stream          Always stream the response body
+        --check-status    Exit with an error status code if the server replies with an error
         --help            Prints help information
     -V, --version         Prints version information
 
 OPTIONS:
     -A, --auth-type <auth-type>              Specify the auth mechanism [possible values: Basic, Bearer]
-    -a, --auth <auth>
+    -a, --auth <USER[:PASS]>                 Authenticate as USER. PASS will be prompted if missing

```

Comment:
I was a bit hesitant to add documentation for _--auth_  since what it takes is determined by the `auth-type` which defaults to `bearer` instead of `basic`.

* when auth-type is bearer (default):
  * token (a shortcut for `authorization:bearer token`)
* when auth-type is basic
  * username:password
  * username (a password prompt will be displayed later)
  * username: for passwordless authentication

Reply:
Aha, I missed that part. I wonder if it would make more sense to split that off into a separate flag? HTTPie's only alternative `auth-method` still works with username and password.

Reply:
>HTTPie's only alternative auth-method still works with username and password.

Could you clarify a bit more?

Reply:
HTTPie has `--auth-type basic` and `--auth-type digest`. But both of them use a username and a password, so `--auth` takes the same kind of argument either way.

`--auth-type bearer` changes how `--auth` works, so maybe it's better to split it off completely into a `--bearer` flag that doesn't interact with `--auth-type`. That would be easier to explain.

You'd have:
- `--auth <USER[:PASS]>`
- `--bearer <TOKEN>`

And no more `--auth-type` (unless e.g. `--auth-type digest` is added).

Reply:
Sounds good. Thanks!

Reply:
Great! Probably something for another PR, for now I've just removed the description again.